firmware/tests/dac_adc_diagnostics/dac_adc_diagnostics.ino

```diff
@@ -55,165 +55,80 @@ static void init_can ( void )
 // measure that the output is what is expected.
-void test_DACS( ) 
+void test_DACS( )
 {
-    uint32_t dac_value;
-    int spoof_signal_a_read = 0;
-    int spoof_signal_b_read = 0;
-    float spoof_a_voltage_read = 0;
-    float spoof_b_voltage_read = 0;
+    uint16_t dac_value;
+    uint16_t dac_val_a;
+    uint16_t dac_val_b;
+
+    uint16_t spoof_a_adc_signal = 0;
+    uint16_t spoof_b_adc_signal = 0;
 
-    float dac_expected_output;
+    float spoof_a_adc_volts = 0.0;
+    float spoof_b_adc_volts = 0.0;
+
+    float dac_expected_output_a = 0.0;
+    float dac_expected_output_b = 0.0;
 
     // energize the relay so we can read the values at the terminal
     digitalWrite( SPOOF_ENGAGE, HIGH );
 
-    for ( dac_value = 0; dac_value < 4096; dac_value = dac_value + 15 )
+    for ( dac_value = 0; dac_value < 4095; dac_value = dac_value + 15 )
     {
-        dac_expected_output = ( 5.0 / 4096.0 ) * dac_value;
-
-        dac.outputA( dac_value );
-        dac.outputB( dac_value );
+        dac_val_a = dac_value;
+        dac_val_b = 4095.0 - dac_value;
 
-        delay( 2000 );
+        dac_expected_output_a = ( 5.0 / 4095.0 ) * dac_val_a;
+        dac_expected_output_b = ( 5.0 / 4095.0 ) * dac_val_b;
 
-        spoof_signal_a_read = analogRead( SPOOF_SIGNAL_A );
-        spoof_signal_b_read = analogRead( SPOOF_SIGNAL_B );
-
-        spoof_a_voltage_read = ( spoof_signal_a_read * 5.0 ) / 1024.0;
-        spoof_b_voltage_read = ( spoof_signal_b_read * 5.0 ) / 1024.0;
-
-        Serial.print( "DAC Value: " );
-        Serial.print( dac_value );
-
-        Serial.print( "\t\t\t\t\tExpected Voltage: " );
-        Serial.println( dac_expected_output,3 );
-
-        Serial.print( "Spoof A Value: " );
-        Serial.print( spoof_signal_a_read );
+        dac.outputA( dac_val_a );
+        dac.outputB( dac_val_b );
 
-        Serial.print( "\tSpoof B Value: " );
-        Serial.print( spoof_signal_b_read );
-
-        Serial.print( "\tSpoof A Voltage: " );      
-        Serial.print( spoof_a_voltage_read, 3 );      
-
-        Serial.print( "\tSpoof B Voltage: " );      
-        Serial.println( spoof_b_voltage_read, 3 );  
+        delay( 2000 );
 
-        Serial.println( "" );    
-    }    
-}
+        spoof_a_adc_signal = analogRead( SPOOF_SIGNAL_A );
+        spoof_b_adc_signal = analogRead( SPOOF_SIGNAL_B );
 
-// to test that the signal interrupt relay functions we can blink two
-// LEDS in an alternating pattering using the the switch relay.
-void test_interrupt_relay( ) 
-{
-    Serial.flush(); 
+        spoof_a_adc_volts = ( spoof_a_adc_signal * 5.0 ) / 1023.0;
+        spoof_b_adc_volts = ( spoof_b_adc_signal * 5.0 ) / 1023.0;
 
-    while( !Serial.available() )
-    {
-        digitalWrite( SPOOF_ENGAGE, LOW );       // start off with the relay in it's depowered state
-        delay( 500 );                            // wait half a second
+        Serial.print( "DAC Value (A): " );
```

Comment:
All serial prints from here down to line 119 need to be removed. 

We decided that the firmware should not use any `Serial.print()` routines.